thirdparty/libart/src/art.c

```diff
@@ -215,15 +265,15 @@ static int check_prefix(const art_node *n, const unsigned char *key, int key_len
 
 /**
  * Checks if a leaf matches
- * @return 0 on success.
+ * @return true if the key is an exact match.
  */
-static int leaf_matches(const art_leaf *n, const unsigned char *key, int key_len, int depth) {
-    (void)depth;
+static bool leaf_matches(const art_leaf *n, const unsigned char *key, int key_len) {
```

Comment:
yeah, it didn't seem like the upstream was very open to making changes in any case, and I've got my eye on a further change to allow holding a reference to our `w_string_t` in the leaves rather than making a copy.